tests/cpp/operator/test_rmsnorm.cu

```diff
@@ -0,0 +1,231 @@
+  workspace = Tensor(workspace.shape(), workspace.dtype());
+  barrier = Tensor(barrier.shape(), barrier.dtype());
+  nvte_rmsnorm_fwd(input.data(), gamma.data(), epsilon, z.data(), rsigma.data(), 0,
+                   prop.multiProcessorCount, workspace.data(), barrier.data());
+
+  // Backward kernel
+  nvte_rmsnorm_bwd(dz.data(), input.data(), rsigma.data(), gamma.data(), dx.data(), dgamma.data(),
+                   dgamma_part.data(), 0, prop.multiProcessorCount, workspace.data(),
+                   barrier.data());
+  workspace = Tensor(workspace.shape(), workspace.dtype());
+  barrier = Tensor(barrier.shape(), barrier.dtype());
+  dgamma_part = Tensor(dgamma_part.shape(), dgamma_part.dtype());
+  nvte_rmsnorm_bwd(dz.data(), input.data(), rsigma.data(), gamma.data(), dx.data(), dgamma.data(),
+                   dgamma_part.data(), 0, prop.multiProcessorCount, workspace.data(),
+                   barrier.data());
+
+  // Reference implementations
+  // use the GPU stats to tighten the tolerances
+  rsigma.to_cpu();
+  float ref_amax;
+  compute_ref_stats(input.cpu_dptr<InputType>(), ref_rsigma.get(), N, H, epsilon);
+  float ref_scale = isFp8Type(otype) ? z.scale() : 1.f;
+  compute_ref_output(input.cpu_dptr<InputType>(), gamma.cpu_dptr<WeightType>(), ref_output.get(),
+                     rsigma.cpu_dptr<float>(), N, H, &ref_amax, ref_scale);
+  compute_ref_backward(dz.cpu_dptr<WeightType>(), input.cpu_dptr<InputType>(),
+                       rsigma.cpu_dptr<float>(), gamma.cpu_dptr<WeightType>(), ref_dx.get(),
+                       ref_dgamma.get(), N, H);
+
+  cudaDeviceSynchronize();
+  auto err = cudaGetLastError();
+  ASSERT_EQ(err, cudaSuccess) << cudaGetErrorString(err);
+
+  auto [atol_amax, rtol_amax] = getTolerances(DType::kFloat32);
+  if (isFp8Type(otype)) {
+    compareResults("amax", z.amax(), ref_amax, atol_amax, rtol_amax);
+    float ref_scale_inv = 1.f / z.scale();
+    compareResults("scale_inv", z.scale_inv(), ref_scale_inv, atol_amax, rtol_amax);
+  }
+
+  auto [atol_stats, rtol_stats] = getTolerances(DType::kFloat32);
+  rtol_stats = 5e-5;
+  compareResults("rsigma", rsigma, ref_rsigma.get(), atol_stats, rtol_stats);
+
+  auto [atol, rtol] = getTolerances(otype);
+  atol = 1e-8;
+  compareResults("output", z, ref_output.get(), atol, rtol);
+
+  double atol_bwd = 5e-6;
+  double rtol_bwd = 1e-4;
+  compareResults("dx", dx, ref_dx.get(), atol_bwd, rtol_bwd);
+  compareResults("dgamma", dgamma, ref_dgamma.get(), atol_bwd, rtol_bwd);
+}
+
+std::vector<std::pair<size_t, size_t>> test_cases = {{2048, 12288}, {768, 1024},
+                                                     {256, 65536},  {128, 6144},
+                                                     {64, 2304},    {229, 541},  // Primes 50, 100
+                                                     {71, 3571},                 // Primes 20, 500
+                                                     {29, 17389}};               // Primes 10, 2000
```

Comment:
Only the `{768,1024}` case will test the tuned kernels. How about:
```suggestion
std::vector<std::pair<size_t, size_t>> test_cases = {{2048, 4096},
                                                     {768, 2048},
                                                     {256, 1024},
                                                     {128, 768},
                                                     {64, 512},
                                                     {173, 409},      // Primes 40, 80
                                                     {71, 3571},      // Primes 20, 500
                                                     {29, 17389}};    // Primes 10, 2000
```

Reply:
Thanks for the suggestion, I will see how to seperate them. Also, I will change the test case to fit our model size, it should be smaller.

Reply:
[132419a037203efa0d767e6cb0e1f81ef521e3f5](https://github.com/NVIDIA/TransformerEngine/pull/45/commits/132419a037203efa0d767e6cb0e1f81ef521e3f5) seperates the `layernorm` and `rmsnorm` implementation. 
[0f6cfff0b992b52e4a16c98ea8aa41ff776e5a12](https://github.com/NVIDIA/TransformerEngine/pull/45/commits/0f6cfff0b992b52e4a16c98ea8aa41ff776e5a12) changes the testcases.